ingestion-beam/src/main/java/com/mozilla/telemetry/schemas/SchemaStore.java

```diff
@@ -131,10 +140,66 @@ private void loadAllSchemas() throws IOException {
     dirs = tempDirs;
   }
 
+  private void loadSchemaAliases() throws IOException {
+    final Set<String> tempDirs = new HashSet<>();
+    final Map<String, T> tempSchemas = new HashMap<>();
+    Multimap<String, String> schemaAliasesMap;
+    try {
+      Multimap<String, String> aliases = ArrayListMultimap.create();
+      if (schemaAliasesLocation != null && schemaAliasesLocation.isAccessible()
+          && !Strings.isNullOrEmpty(schemaAliasesLocation.get())) {
+        Metadata metadata = FileSystems.matchSingleFileSpec(schemaAliasesLocation.get());
+        InputStream configInputStream = Channels
+            .newInputStream(FileSystems.open(metadata.resourceId()));
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        SchemaAliasingConfiguration aliasingConfig = objectMapper.readValue(configInputStream,
+            SchemaAliasingConfiguration.class);
+
+        aliasingConfig.aliases().forEach((e) -> {
+          aliases.put(e.base(), e.alias());
+        });
+      }
+      schemaAliasesMap = aliases;
+    } catch (IOException e) {
+      throw new IOException("Exception thrown while fetching from configured schemaAliasesLocation",
+          e);
+    }
+
+    dirs.forEach(dir -> {
+      tempDirs.add(dir);
+      tempDirs.addAll(schemaAliasesMap.get(dir));
+    });
+
+    schemas.forEach((schemaName, schema) -> {
+      tempSchemas.put(schemaName, schema);
+
+      String[] components = schemaName.split("/");
+      String namespaceWithDocType = components[0] + "/" + components[1];
+      // if aliases are defined for current `schemaName`, assemble new aliasing path and put it to
+      // the map with original schema
+      if (schemaAliasesMap.containsKey(namespaceWithDocType)) {
+        for (String namespaceWithDocTypeAlias : schemaAliasesMap.get(namespaceWithDocType)) {
+          String docTypeWithVersion = components[components.length - 1];
+          String versionAndSuffix = docTypeWithVersion.substring(docTypeWithVersion.indexOf("."));
+
+          String docTypeAlias = namespaceWithDocTypeAlias.split("/")[1];
+          String docTypeWithVersionAlias = docTypeAlias + versionAndSuffix;
+
+          tempSchemas.put(namespaceWithDocTypeAlias + "/" + docTypeWithVersionAlias, schema);
+        }
+      }
+    });
+
+    schemas = tempSchemas;
+    dirs = tempDirs;
+  }
```

Comment:
This is a pretty elegant approach to merging in aliased schemas to the existing schema map leaving the existing interface for looking up schemas unchanged. The one drawback I see to this is that it doesn't give ParsePayload any way of knowing that it was a glean schema being used for validation.

We had discussed previously adding "validation_namespace" and "validation_schema" attributes to the PubsubMessage in the case that an alias was followed to determine a schema to validate against. Those attributes could then be used by the Debug View pipeline to filter out any non-Glean messages. Does that still seem useful? If so, we may need to change the interface here so that we can return the name of the underlying schema returned.

Reply:
> We had discussed previously adding "validation_namespace" and "validation_schema" attributes to the PubsubMessage in the case that an alias was followed to determine a schema to validate against.

I'm sorry, I'm not sure I follow. If we go with `incoming.t.m.o/glean+product/glean+doc/1/{uuid}`, these attributes can be added to metadata earlier in the processing chain (say, `ParseUri`?). Then `SchemaStore` wouldn't need to be aware of aliasing and we could just revert its changes I'm introducing here.
Does this make sense?

Reply:
Yes, that makes sense. For the time being, then, the republisher topic for debug pings won't have any metadata that distinguishes glean pings from others and you'll have to look inside the payload to determine if something follows the glean schema. If that's okay for your purposes, then it's fine by me.

Reply:
> republisher topic for debug pings won't have any metadata that distinguishes glean pings from others

I think that's OK - I can look at `document_namespace`+`document_type` and validate against a list of products that I'm interested in (I could even reuse aliasing configuration introduced in this PR for that).